python/semantic_kernel/connectors/ai/mistral_ai/settings/mistral_ai_settings.py

```diff
@@ -0,0 +1,32 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+from typing import ClassVar
+
+from pydantic import SecretStr
+
+from semantic_kernel.kernel_pydantic import KernelBaseSettings
+
+
+class MistralAISettings(KernelBaseSettings):
+    """MistralAI model settings.
+
+    The settings are first loaded from environment variables with the prefix 'MISTRAL_'. If the
+    environment variables are not found, the settings can be loaded from a .env file with the
+    encoding 'utf-8'. If the settings are not found in the .env file, the settings are ignored;
+    however, validation will fail alerting that the settings are missing.
+
+    Optional settings for prefix 'MISTRAL_' are:
+    - api_key: SecretStr - MISTRAL API key, see https://console.mistral.ai/api-keys
+        (Env var MISTRAL_API_KEY)
+    - chat_model_id: str | None - The The Mistral AI chat model ID to use see https://docs.mistral.ai/getting-started/models/.
+        (Env var MISTRAL_CHAT_MODEL_ID)
+    - embedding_model_id: str | None - The Mistral AI embedding model ID to use, for example, mistral-embed.
+        (Env var MISTRAL_EMBEDDING_MODEL_ID)
+    - env_file_path: str | None - if provided, the .env settings are read from this file path location
+    """
+
+    env_prefix: ClassVar[str] = "MISTRAL_"
+
+    api_key: SecretStr
+    chat_model_id: str | None = None
+    embedding_model_id: str | None = None
```

Comment:
I see the `embedding_model_id` settings param here. Should we remove it until the embedding support has been added?

Reply:
Thats a good idea, i am wundering if it makes sense, to probably split the work into 3 Pull Requests :
1. ChatCompletion Streaming & Non Streaming
2. Embbedding Integration
3. Function Calling Integration

Otherwise the Request will be very big, i think it's very big already.